### PR TITLE
Docs: Update query caching support

### PR DESCRIPTION
**Pull request by AgnesToulet**
**What this PR does / why we need it**:
Update the sentence about data sources that work with query caching.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/sources/enterprise/query-caching.md [hunk 1]
```diff
@@ -15,7 +15,7 @@ weight: 300
 
 When query caching is enabled, Grafana temporarily stores the results of data source queries. When you or another user submit the exact same query again, the results will come back from the cache instead of from the data source (like Splunk or ServiceNow) itself.
 
-Query caching works for all backend data sources, and queries sent through the data source proxy. You can enable the cache globally and configure the cache duration (also called Time to Live, or TTL).
```
Comment (kminehart) [inline docs/sources/enterprise/query-caching.md:18]: I don't see a reason that it wouldn't work for all backend data sources. The only data sources that send requests through the data source proxy are frontend data sources, right?

Comment (AgnesToulet) [reply docs/sources/enterprise/query-caching.md:18]: But is there an easy way for the user to know if a data source is backend or frontend? If so, we should document it; if not, we shouldn't keep that sentence in the documentation IMO.

Comment (kminehart) [reply docs/sources/enterprise/query-caching.md:18]: The only way to tell without looking at the code is to see if the caching tab is there I think.

Comment (gabor) [reply docs/sources/enterprise/query-caching.md:18]: sorry to jump in into this discussion, but "backend data sources, and queries sent through the data source proxy" is nearly everyhing. it's the same as "all data sources, except those that use access=browser", if i remember correctly.

(i know nothing about how query-caching works 😄  , i'm only talking about what data-source-proxy means here)

Comment (kminehart) [reply docs/sources/enterprise/query-caching.md:18]: Is it possible to have a backend datasource with "access=browser"? That doesn't sound right to me.

Comment (gabor) [reply docs/sources/enterprise/query-caching.md:18]: i don't think it is possible. it should be like this:
- backend-datasource: the request goes browser -> datasource_specific_go_code -> database
- frontend-datasource:
    - access=server: the request goes browser -> generic_grafana_go_code -> database
    - access=browser: the request goes browser -> dateabase

Comment (kminehart) [reply docs/sources/enterprise/query-caching.md:18]: right, so to me the text here is still true. Caching only works on backend data sources. The original mistake was that the documentation said it was possible to cache requests through the data source proxy but that was actually wrong. We are not able to cache requests that go through the proxy.

**Review by kminehart — APPROVED** [review]

**Pull request merged**